Code/src/main/java/nl/utwente/viskell/haskell/env/CatalogFunction.java

```diff
@@ -33,6 +37,31 @@
         this.isCommon = isCommon;
     }
 
+    @Override
+    public Map<String, Object> toBundleFragment() {
+        Map<String, Object> bundleFragment = new HashMap<>();
+        bundleFragment.put(Bundleable.KIND, this.getClass().getSimpleName());
+        bundleFragment.put("name", name);
+        bundleFragment.put("signature", signature.toString());
+        bundleFragment.put("category", category);
+        bundleFragment.put("documentation", documentation);
+        bundleFragment.put("isConstructor", isConstructor);
+        bundleFragment.put("isCommon", isCommon);
```

Comment:
lookupFun(9 returns FunctionInfo not a CatalogFunction, so I guess I'll have to make a new constructor for CatalogFunction() that accepts a FunctionInfo - I need to read the code more...